internal/core/layout.rs

```diff
@@ -305,23 +305,28 @@ mod grid_internal {
         assert_eq!(my_items[2].size, 100.);
     }
 
-    /// Create a vector of LayoutData for an array of GridLayoutCellData
+    /// Create a vector of LayoutData (e.g. one per row if Vertical) for an array of GridLayoutCellData (one per cell)
+    /// Used by both solve_grid_layout() and grid_layout_info() (each with their own copy of the GridLayoutCellData slice)
     pub fn to_layout_data(
         data: &[GridLayoutCellData],
         spacing: Coord,
         size: Option<Coord>,
-    ) -> Vec<LayoutData> {
+        orientation: Orientation,
+    ) -> (Vec<LayoutData>, Vec<GridLayoutCellData>) {
+        let mut cells: Vec<GridLayoutCellData> = data.to_vec();
```

Comment:
slightly concerned about the extra allocation here at runtime. But I hope this is fine and we can optimize later if needed.

Reply:
Ah, maybe I could change the generated code to generate a Vec (that I can mutate) instead of
 `&sp::GridLayoutData { cells: sp::Slice::from_slice(&[ ... ]) )` ?